terraform/byo-vpc/main.tf

```diff
@@ -97,7 +97,7 @@ module "redis" {
     from_port   = 0
     to_port     = 65535
     protocol    = "tcp"
-    cidr_blocks = ["10.0.0.0/8"]
+    cidr_blocks = [for s in data.aws_subnet.redis : s.cidr_block]
```

Comment:
Has this been tested?  I don't think it does what we want.  I think this rule should be to allow fleet in rather than be the redis subnets.  My thoughts were that we needed a separate subnet group like this:

```
data "aws_subnet" "allowed" {
  for_each = toset(var.redis_config.allowed_subnets)
  id       = each.value
}
```

By default we have the `module.main` pass in the subnets for fleet obtained by the `private` subnets in the module.  We won't be able to have defaults  at the `byo-vpc` layer, but we can make the variable required and need at least one subnet `type = list(string)` or whtaever.

Then we'll need to probably spin up a new full cloud testing it like spin up the template example in cloud or apply it to dogfood temporarily or whatnot.

Reply:
I'm spinning up an environment this morning to give it a go.

Reply:
plan after revised changes:

```
  # module.main.module.byo-vpc.module.redis.module.aws_security_group.aws_security_group_rule.keyed["extra[0]"] must be replaced
+/- resource "aws_security_group_rule" "keyed" {
      ~ cidr_blocks              = [ # forces replacement
          - "10.0.0.0/8",
          + "10.10.1.0/24",
          + "10.10.2.0/24",
          + "10.10.3.0/24",
        ]
      ~ id                       = "sgrule-1866165834" -> (known after apply)
      ~ security_group_rule_id   = "sgr-0d01dac15808a530a" -> (known after apply)
      + source_security_group_id = (known after apply)
        # (8 unchanged attributes hidden)
    }

```

Reply:
looks good.  Once apply is done and validated wherever you are testing, will be glad to mark everything as approved.

Reply:
Testing in cloud environment I can confirm redis is still reachable by the application and I can issue things like live queries.